### PR TITLE
[DNM] Brand - fix P2 a11y bug limiting iOS Voice Control

### LINKED ISSUE #5514: a11y P2 - Voice control (iOS) - Brand bug
**Describe the bug**
The BBC Brand link in the banner does not work when you say 'Tap BBC News Pidgin' on https://www.bbc.com/pidgin When you say 'Show names' and 'Show numbers' the brand link is also not displayed. Note, not a P1 as you can access the home page via the 'Home' link in the navigation. 

I've tested removing the role text attribute and the lang attribute from the visually hidden link text and this appears not to affect it. This appears to be a bug related to the visually hidden CSS. Though note, you can say 'tap video' and this works for promos with videos on the url and this is visually hidden text, maybe that's because this element also has displayed text with it? I've tested adding some displayed text to the brand link and this fixes it, though not a solution we can use.

Also tested using a SVG title element instead of VH text, this resulted in the brand link being available when you say 'show numbers' but not the other navigation methods listed.

Note the brand logo works on a News Optimo article https://www.bbc.co.uk/news/articles/clldg965yzjo with exactly the same VH CSS... This is an odd bug... Please investigate further. Re-test this before investigating to check it's still an issue.

Test page from initial investigations: http://2bitstudio.co.uk/code-tests/world-service/brand-voice-control-test.html

**To Reproduce**
1. Open Safari and go to https://www.test.bbc.com/pidgin/media-43768328?renderer_env=live
2. Say 'Tap BBC News Pidgin'
3. The link is not activated
4. Say 'Show names' 
5. The link is not listed
6. Say 'Show numbers'
7. The link is not numbered

**Expected behaviour**
1. Go to https://www.test.bbc.com/pidgin/media-43768328?renderer_env=live
2. Say 'Tap BBC News Pidgin'
3. The link should be activated
4. Say 'Show names' 
5. The link should be listed
6. Say 'Show numbers'
7. The link should be numbered

**Testing notes**

- [x] This bug fix is expected to need manual testing.

### PR DESCRIPTION
**Pull request by simonsinclair**
Resolves https://github.com/bbc/simorgh/issues/5514.

### To do:
Please familiarise yourself with the thread below, and see https://github.com/bbc/psammead/pull/3131#issuecomment-590332361.
1. Try adding back in the lang="en-GB" (only) and see if a Voice Control tests pass.

### Overall change:
Simplify visually hidden text.

### Code changes:

- Simplify visually hidden text
  - Remove nested spans
  - Remove [problematic](https://incl.ca/roletext-presently-kinda-not-thing-sorta/) ARIA role
  - Prevent React comment `<!-- -->`
- Update tests
- Update snapshots
- Set a sane `serviceLocalisedNameInput` default value in Storybook
- Bump patch version
- Add changelog entry

#### Brand - RTL, Arabic

IMG_8363

#### Brand - Default

IMG_8365

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [x] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-brand/src/index.test.jsx [hunk 1]
```diff
@@ -78,114 +78,76 @@ describe('Brand', () => {
...
+      <ScriptLink
+        script={latin}
+        service="serbian"
+        href="https://www.bbc.com/serbian/lat"
+      >
+        Lat
+      </ScriptLink>
+    );
+
+    const skipLink = (
```
Comment (mmcculloch) [inline packages/components/psammead-brand/src/index.test.jsx]: Should this be called `skipLinkComponent`? Just for consistency with `scriptLinkComponent`.

**Review by mmcculloch — APPROVED** [review]

**Comment by greenc05** [comment]
@simonsinclair When I tested this on my test page last week, with the same changes this didn't fix it, are we 100% this fixes this issue?

**Comment by greenc05** [comment]
Also unsure about removing the lang attribute... If we go with this, we should add an a11y note to say why we are no longer using this.

File: packages/components/psammead-brand/src/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -122,15 +122,8 @@ exports[`Brand should render correctly with link not provided 1`] = `
...
-      role="text"
     >
-      <span
-        lang="en-GB"
-      >
-        Default Brand Name
-      </span>
-      , 
-      Service
+      Default Brand Name Service
```
Comment (greenc05) [inline packages/components/psammead-brand/src/__snapshots__/index.test.jsx.snap]: Should be "Default Brand Name, Service"

**Review by andrewscfc — COMMENTED** [review]
LGTM, I can't check the changes have the desired effect myself so I'll leave another approval to someone who has access to an iphone!

**Comment by HarveyPeachey** [comment]
> @simonsinclair When I tested this on my test page last week, with the same changes this didn't fix it, are we 100% this fixes this issue?

Which test page were you using?

**Review by HarveyPeachey — COMMENTED** [review]
Weird I'm not seeing it in storybook, maybe it's because I'm using the chromatic build?

Image from iOS

File: packages/components/psammead-brand/src/index.test.jsx (hunk 1 above)
Comment (simonsinclair) [reply packages/components/psammead-brand/src/index.test.jsx]: These tests weren't added in this PR, but any opportunity to improve consistency. 😄

File: packages/components/psammead-brand/src/__snapshots__/index.test.jsx.snap (hunk 2 above)
Comment (simonsinclair) [reply packages/components/psammead-brand/src/__snapshots__/index.test.jsx.snap]: Thanks. I've fixed this.

**Comment by simonsinclair** [comment]
> When I tested this on my test page last week, with the same changes this didn't fix it, are we 100% this fixes this issue?

@greenc05 yeah, they appear to fix the issue. Let me know if you need any help testing them.


> Also unsure about removing the lang attribute... If we go with this, we should add an a11y note to say why we are no longer using this.

@greenc05 this got culled, because it was adding (unnecessary?) complexity to the markup iOS Voice Control was struggling with. Do we need to specifically define the language of a trademark like "BBC News"? I couldn't find any examples of where we do this.

---

> Weird I'm not seeing it in storybook, maybe it's because I'm using the chromatic build?

@HarveyPeachey, this bug only affects versions of the Brand component that use a link. E.G. `components-brand--with-brand-link` on Storybook. Did you have this selected?

**Review by 12 — COMMENTED** [review]
Looks good to me, not tested however.

**Review by HarveyPeachey — APPROVED** [review]
LGTM/SGTM tested and it works :)

**Comment by greenc05** [comment]
@simonsinclair yes we always add a lang attribute where we have control, if the content isn't in the language of the page e.g. the 'BBC News' part of the brand is in english, you can see an example of this on the h1 on the front pages. This will now mean that BBC News in the brand will announce on a screen reader with an accent of the language of the page rather than with the correct English accent... I'm not happy removing this, though if this is the only fix, then we can go ahead and we should add an a11y note to say why we are no longer doing this please. (Note I haven't tested your code)

Thanks for adding the comma back.

**Comment by greenc05** [comment]
Ok, so... Technically the user can access the home link via the navigation, if after some investigations if we can't fix this bug and keep the lang attribute on the english text and role text to stop text splitting with VoiceOver, then I think we should close this issue and I'll report the issue to Apple...

**Comment by simonsinclair** [comment]
Closing, because after further investigation with @FK78, I can confirm that _any_ nested elements within the visually hidden `span` causes iOS' Voice Control to fail. It is not caused by the addition of a `lang` attribute.

@greenc05's preference is to maintain the `lang` attribute for screen readers, and will raise a bug report with Apple about the above issue.

**Comment by greenc05** [comment]
This bug has been reported to Apple at  accessibility@apple.com "If a link has no visually displayed text e.g. all the link text is visually hidden with CSS and this contains more than one nested HTML element, then the link is not available to Voice Control, either by saying 'Tap X' or via saying 'Show names' or 'Show numbers'."

**Pull request closed without merging**